src/Core/Compiler/NodeConverter.php

```diff
@@ -367,9 +375,34 @@ protected function convertExpressionNode(ExpressionNodeInterface $node)
      */
     protected function convertBooleanNode(BooleanNode $node)
     {
-        $stack = $this->convertArrayNode(new ArrayNode($node->getStack()));
-        $initializationPhpCode = '// Rendering Boolean node' . chr(10);
-        $initializationPhpCode .= $stack['initialization'] . chr(10);
+        $booleanStack = $node->getStack();
+
+        // Quick decisions: if there is only one node and it can be determined to be a string/numeric/boolean already
+        // then use it directly.
+        if (count($booleanStack) === 1) {
+            $compiledOnlyNode = $this->convert($booleanStack[0]);
+            // Execution without possible quotation marks, then converted to lowercase for easier comparison.
+            $execution = trim($compiledOnlyNode['execution'], '"\'\\');
+            $lowercaseExecution = strtolower($execution);
+            if (is_numeric($execution) || is_bool($execution)) {
+                $execution = $execution ? 'true' : 'false';
+            } elseif ($lowercaseExecution === 'false' || empty($execution)) {
```

Comment:
Doing the empty check first should be faster IMO.

Reply:
Hmm I don't think that's the case. An explicit comparison checks for one specific case of equality whereas the empty has to check for many. But you may have a point here that these should go *before* the is-numeric and is-bool stuff!

Reply:
It also doesn't matter a whole lot because this code is only for compiling the template so it's very infrequent ;)